projects/Billing_system/test_BillAppBillArea.py

```diff
@@ -0,0 +1,248 @@
+        assert bill_app.txtarea.get("1.0", "end-1c") == expected_output
+
+    @pytest.mark.regression
+    @pytest.mark.valid
+    def test_grocery_products_purchased(self, bill_app):
+        bill_app.rice.set(10)
+        bill_app.food_oil.set(20)
+        bill_app.wheat.set(30)
+        bill_app.daal.set(40)
+        bill_app.flour.set(50)
+        bill_app.maggi.set(60)
+        bill_app.bill_area()
+        expected_output = f"\n Rice\t\t10\t\t{bill_app.g_r_p}\n Food Oil\t\t20\t\t{bill_app.g_f_o_p}\n Wheat\t\t30\t\t{bill_app.g_w_p}\n Daal\t\t40\t\t{bill_app.g_d_p}\n Flour\t\t50\t\t{bill_app.g_f_p}\n Maggi\t\t60\t\t{bill_app.g_m_p}"
+        assert bill_app.txtarea.get("1.0", "end-1c") == expected_output
+
+    @pytest.mark.regression
+    @pytest.mark.valid
+    def test_cold_drinks_products_purchased(self, bill_app):
+        bill_app.sprite.set(10)
+        bill_app.limka.set(20)
+        bill_app.mazza.set(30)
+        bill_app.coke.set(40)
+        bill_app.fanta.set(50)
+        bill_app.mountain_duo.set(60)
+        bill_app.bill_area()
+        expected_output = f"\n Sprite\t\t10\t\t{bill_app.c_d_s_p}\n Limka\t\t20\t\t{bill_app.c_d_l_p}\n Mazza\t\t30\t\t{bill_app.c_d_m_p}\n Coke\t\t40\t\t{bill_app.c_d_c_p}\n Fanta\t\t50\t\t{bill_app.c_d_f_p}\n Mountain Duo\t\t60\t\t{bill_app.c_m_d}"
+        assert bill_app.txtarea.get("1.0", "end-1c") == expected_output
+
+    @pytest.mark.regression
+    @pytest.mark.valid
+    def test_taxes_applied(self, bill_app):
+        bill_app.medical_tax.set("10.0")
+        bill_app.grocery_tax.set("20.0")
+        bill_app.cold_drinks_tax.set("30.0")
+        bill_app.bill_area()
+        expected_output = f"\n Medical Tax\t\t\t10.0\n Grocery Tax\t\t\t20.0\n Cold Drinks Tax\t\t\t30.0"
+        assert bill_app.txtarea.get("1.0", "end-1c") == expected_output
+
+    @pytest.mark.regression
+    @pytest.mark.valid
+    def test_total_bill_calculation(self, bill_app):
+        bill_app.sanitizer.set(10)
+        bill_app.mask.set(20)
+        bill_app.hand_gloves.set(30)
+        bill_app.dettol.set(40)
+        bill_app.newsprin.set(50)
+        bill_app.thermal_gun.set(60)
+        bill_app.rice.set(10)
+        bill_app.food_oil.set(20)
+        bill_app.wheat.set(30)
+        bill_app.daal.set(40)
+        bill_app.flour.set(50)
+        bill_app.maggi.set(60)
+        bill_app.sprite.set(10)
+        bill_app.limka.set(20)
+        bill_app.mazza.set(30)
+        bill_app.coke.set(40)
+        bill_app.fanta.set(50)
+        bill_app.mountain_duo.set(60)
+        bill_app.medical_tax.set("10.0")
+        bill_app.grocery_tax.set("20.0")
+        bill_app.cold_drinks_tax.set("30.0")
+        bill_app.bill_area()
+        assert bill_app.txtarea.get("1.0", "end-1c").endswith(f"\n Total Bil:\t\t\t Rs.{bill_app.total_bill}")
+
+class BillApp:
```

Comment:
class code is redeclared in test file
and the instance is missing for the test functions above

Reply:
pytest fixture could be used